base/sparse/higherorderfns.jl

```diff
@@ -148,10 +148,13 @@ ambiguityfunnel{Tf}(f::Tf, x, y) = _aresameshape(x, y) ? _noshapecheck_map(f, x,
 broadcast(::typeof(+), x::SparseVector, y::SparseVector) = ambiguityfunnel(+, x, y) # base/sparse/sparsevectors.jl:1266
 broadcast(::typeof(-), x::SparseVector, y::SparseVector) = ambiguityfunnel(-, x, y) # base/sparse/sparsevectors.jl:1266
 broadcast(::typeof(*), x::SparseVector, y::SparseVector) = ambiguityfunnel(*, x, y) # base/sparse/sparsevectors.jl:1266
-function broadcast!(::typeof(identity), C::SparseMatrixCSC, A::SparseMatrixCSC) # from #17623, loc?
+# specializations to improve performance for C .= A
+function broadcast!(::typeof(identity), C::SparseVector, A::SparseVector)
```

Comment:
Could you also add an `@boundscheck` so people can remove shape checking when they already checked that before hand or know in advance the arrays are of the same size?